src/core/patterns.js

```diff
@@ -152,3 +152,78 @@ foam.CLASS({
     }
   ]
 });
+
+/** Flyweight classes create instances that can create flyweight copies.
```

Comment:
Yes, but it's flyweight in that "If a flyweight pattern is what's appropriate, use this instead".

A Java implementation would look very different on the output side, but ideally we could generate it  from the same models.